clang-tools/CMakeLists.txt

```diff
@@ -1,23 +1,45 @@
-cmake_minimum_required(VERSION 2.8.4)
-project(clang-tools)
+cmake_minimum_required ( VERSION 3.10.2 )
```

Comment:
this is way too new a version of cmake to make mandatory.

Reply:
Version 3.10.2 is what comes with ubuntu bionic, which is the only OS release we are testing the semantics on.

In general, `cmake` has been dramatically improved in these later versions, and I'd argue it's okay to demand this new a version (this is an easy to obtain tool, even without a package manager). I could, however, see your point if you told me that the version installed on debian stretch is 3.7.2.

Reply:
We definitely should support at least as far back as Ubuntu 16.04 which has CMake 3.5.

Reply:
Sounds reasonable. I double-checked that I am not using anything that is not available in 3.5.